File: CS/SynchronizeEditValuesInEditForm_MVVM/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using DevExpress.Xpf.Grid;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SynchronizeEditValuesInEditForm_MVVM {
    public class DataItem {
        public int Amount { get; set; }

        public int Price { get; set; }

        public int PositionValue { get => Price * Amount; }

        public bool CanEdit { get; set; } = true;

        public DataItem(Random random) {
            Amount = random.Next(1, 10);
            Price = random.Next(100, 1000);
        }
    }
    class MainViewModel : ViewModelBase {
        public ObservableCollection<DataItem> Items { get; }

        public MainViewModel() {
            Items = new ObservableCollection<DataItem>(GetData(10));
        }

        static IEnumerable<DataItem> GetData(int amount) {
            var random = new Random();
            return Enumerable.Range(0, amount).Select(i => new DataItem(random));
        }

        [Command]
        public void SynchronizeValues(CellValueChangedArgs args) {
            var editFormArgs = (CellValueChangedInEditFormArgs)args;
            if(editFormArgs == null || (args.FieldName != nameof(DataItem.Price) && args.FieldName != nameof(DataItem.CanEdit))) {
                return;
            }

            if(args.FieldName == nameof(DataItem.CanEdit)) {
                editFormArgs.CellEditors.FirstOrDefault(x => x.FieldName == "Price").ReadOnly = !bool.Parse(args.Value.ToString());
                return;
            }

            var positionValueData = editFormArgs.CellEditors.First(d => d.FieldName == nameof(DataItem.PositionValue));
            var amountData = editFormArgs.CellEditors.First(d => d.FieldName == nameof(DataItem.Amount));

            int price = 0;

            int.TryParse((string)args.Value, out price);
            positionValueData.Value = (int)amountData.Value * price;
        }

        [Command]
        public void InitializeEditing(RowEditStartingArgs args) {
            args.CellEditors.FirstOrDefault(x => string.Equals(x.FieldName, "Price")).ReadOnly = !(bool)args.CellEditors.FirstOrDefault(x => string.Equals(x.FieldName, "CanEdit")).Value;
        }
    }
}

File: VB/SynchronizeEditValuesInEditForm_MVVM/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace SynchronizeEditValuesInEditForm_MVVM

    Public Class DataItem

        Public Property Amount As Integer

        Public Property Price As Integer

        Public ReadOnly Property PositionValue As Integer
            Get
                Return Price * Amount
            End Get
        End Property

        Public Property CanEdit As Boolean = True

        Public Sub New(ByVal random As Random)
            Amount = random.Next(1, 10)
            Price = random.Next(100, 1000)
        End Sub
    End Class

    Friend Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property Items As ObservableCollection(Of DataItem)

        Public Sub New()
            Items = New ObservableCollection(Of DataItem)(GetData(10))
        End Sub

        Private Shared Function GetData(ByVal amount As Integer) As IEnumerable(Of DataItem)
            Dim random = New Random()
            Return Enumerable.Range(0, amount).[Select](Function(i) New DataItem(random))
        End Function

        <Command>
        Public Sub SynchronizeValues(ByVal args As CellValueChangedArgs)
            Dim editFormArgs = CType(args, CellValueChangedInEditFormArgs)
            If editFormArgs Is Nothing Then
                Return
            End If

            If Equals(args.FieldName, NameOf(DataItem.CanEdit)) Then
                Dim priceData = editFormArgs.CellEditors.FirstOrDefault(Function(x) Equals(x.FieldName, NameOf(DataItem.Price)))
                priceData.[ReadOnly] = Not Boolean.Parse(args.Value.ToString())
                Return
            End If

            If Equals(args.FieldName, NameOf(DataItem.Price)) Then
                Dim positionValueData = editFormArgs.CellEditors.First(Function(d) Equals(d.FieldName, NameOf(DataItem.PositionValue)))
                Dim amountData = editFormArgs.CellEditors.First(Function(d) Equals(d.FieldName, NameOf(DataItem.Amount)))
                Dim price As Integer = 0
                Call Integer.TryParse(CStr(args.Value), price)
                positionValueData.Value = CInt(amountData.Value) * price
            End If
        End Sub

        <Command>
        Public Sub InitializeEditing(ByVal args As RowEditStartingArgs)
            Dim priceData = args.CellEditors.FirstOrDefault(Function(x) Equals(x.FieldName, NameOf(DataItem.Price)))
            Dim canEditData = args.CellEditors.FirstOrDefault(Function(x) Equals(x.FieldName, NameOf(DataItem.CanEdit)))
            priceData.[ReadOnly] = Not CBool(canEditData.Value)
        End Sub
    End Class
End Namespace
